fprettify: init at 0.3.7

FabianGD opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

This PR adds the fprettify Formatter for Fortran sources, which is a very helpful tool when working with Fortran.

(I also add myself as the maintainer of the package, is it ok to do this in the same PR?)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

figsoda commented on pkgs/development/tools/fprettify/default.nix:
```diff
@@ -0,0 +1,28 @@
+{ lib, python3Packages, fetchFromGitHub }:
```
```suggestion
{ lib, python3Packages, fetchFromGitHub, stdenvNoCC }:
```

figsoda commented on pkgs/development/tools/fprettify/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+    # The checks use subprocess.py, which fails with a FileNotFoundError
+    # on Linux. On macOS, it works just fine.
+    doCheck = false;
```
```suggestion
    doCheck = stdenvNoCC.isDarwin;
```

FabianGD replied:
This is not necessary anymore, thanks to your fix. :)

figsoda reviewed (changes requested):
Squash commits to
`maintainers: add fabiangd`
`fprettify: init at 0.3.7`

figsoda commented:
This fixes tests on linux for me
```nix
  preConfigure = ''
    patchShebangs fprettify.py
  '';
```

figsoda commented:
Also, use 2 spaces for indentation

FabianGD committed:
maintainer: add fabiangd

FabianGD commented:
> This fixes tests on linux for me
> 
> ```nix
>   preConfigure = ''
>     patchShebangs fprettify.py
>   '';
> ```

I tried that: Great, it also fixes it for me.

figsoda commented on pkgs/development/tools/fprettify/default.nix:
```diff
@@ -0,0 +1,30 @@
+    configargparse
+  ];
+
+  doCheck = true;
```
```suggestion
```

this is the default

r-rmcgibbo commented:
Result of `nixpkgs-review pr 134423` at 05bab87c run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>fprettify</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `stdenvNoCC`.
Near pkgs/development/tools/fprettify/default.nix:1:8:
```
  |
1 | { lib, stdenvNoCC, python3Packages, fetchFromGitHub }:
  |        ^
```
</li>
  </ul>
</details>

FabianGD committed:
fprettify: init at 0.3.7

FabianGD commented:
I added your requested changes, including those by @r-rmcgibbo. Furthermore, I changed the location of fprettify in the `all-packages.nix` to a more sensible place (in the same section as I found `black`).

figsoda commented:
Result of `nixpkgs-review pr 134423` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fprettify</li>
  </ul>
</details>
